src/analysis/normed/group/pointwise.lean

```diff
@@ -48,46 +48,157 @@ begin
   ... ≤ Rs + Rt : add_le_add (hRs x hx) (hRt y hy)
 end
 
-@[simp] lemma singleton_add_ball (x y : E) (r : ℝ) :
-  {x} + ball y r = ball (x + y) r :=
-by simp only [preimage_add_ball, image_add_left, singleton_add, sub_neg_eq_add, add_comm y x]
+variables {ε δ : ℝ} {s : set E}
 
-@[simp] lemma ball_add_singleton (x y : E) (r : ℝ) :
-  ball x r + {y} = ball (x + y) r :=
-by simp [add_comm _ {y}, add_comm y]
+lemma ennreal.of_real_of_nonpos {r : ℝ} (hr : r ≤ 0) : ennreal.of_real r = 0 :=
+congr_arg coe $ real.to_nnreal_of_nonpos hr
 
-lemma singleton_add_ball_zero (x : E) (r : ℝ) :
-  {x} + ball 0 r = ball x r :=
-by simp
+lemma thickening_of_nonpos (hδ : δ ≤ 0) (s : set E) : thickening δ s = ∅ :=
+eq_empty_iff_forall_not_mem.2 $ λ x hx,
+  (ennreal.of_real_of_nonpos hδ).not_gt $ bot_le.trans_lt hx
```

Comment:
```suggestion
lemma thickening_of_nonpos {α : Type*} [pseudo_emetric_space α] (hδ : δ ≤ 0) (s : set α) :
  thickening δ s = ∅ :=
eq_empty_iff_forall_not_mem.2 $ λ x hx,
  (ennreal.of_real_eq_zero.mpr hδ).not_gt $ bot_le.trans_lt hx
```

This holds in the generality in which thickenings are defined.

Reply:
Done in #13380